File: backend/routers/plans.py
from fastapi import APIRouter

from schemas.event import EventInput, GeneratedPlan
from services.plan_service import generate_plan

router = APIRouter(prefix="/plans", tags=["plans"])


@router.post("/generate", response_model=GeneratedPlan)
def create_plan(event: EventInput) -> GeneratedPlan:
    return generate_plan(event)

File: frontend/src/api/plans.ts
import type { EventInput, GeneratedPlan } from '../types/plan'

const API_BASE = 'http://127.0.0.1:8000'

export async function generatePlan(event: EventInput): Promise<GeneratedPlan> {
  const response = await fetch(`${API_BASE}/plans/generate`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(event),
  })

  if (!response.ok) {
    const text = await response.text()
    throw new Error(text || 'Failed to generate plan')
  }

  return response.json()
}
